### ISSUE TITLE
Remove bundled emulators for the 3DS where there exists a better option

### ISSUE THREAD

**Issue by lifehackerhansol** [DS-Homebrew/TWiLightMenu#1644]
**Is your feature request related to a problem? Please describe.**
This is specific to the 3DS. There isn't much one can do with a DSi with this.

The 3DS already has its fair share of emulators, notably snes9x, VirtuaNES, GBC, etc.
If a well-known emulator exists for the console, I do not think this is required to be added to TWiLight for 3DS.

**Describe the solution you'd like**
Take note of emulators that already have a 3DS-mode alternative, and remove them from the 3DS install package.

**Describe alternatives you've considered**
The only other one I can think of is having some sort of auto-package website where you can select what you want. This idea was brought up at one point over on Discord but nothing came out of it.

**Comment by eku** [comment]
Will it then be possible to launch the emulators installed on the 3DS from TWL++ instead of the emulators included with TWL++?

**Comment by lifehackerhansol** [comment]
I don't think that's possible, even with RTcom. 

I think the best alternative for people who want to load emus from TWiLight's UI for whatever reason should add them manually, I don't like the idea of it being bundled by default when there are obvious better alternatives that take full use of the 3DS hardware.

**Comment by eku** [comment]
It would at least be an improvement if the emulators were downloadable in a separate package instead of the 3DS TWL++ package. What do you think about that?

**Comment by Kaisaan** [comment]
You could just use [Universal-Updater](https://universal-team.net/projects/universal-updater) for that.

**Comment by lifehackerhansol** [comment]
Funny enough, Universal Updater didn't even cross my mind until @Kaisaan mentioned that. That is a perfect solution for a separate package.

**Comment by lifehackerhansol** [comment]
> This issue is already mentioned here: #1204

I am aware of the issue. I wanted a separate one _solely_ for the emulators, while that issue deals with other general things.

**Comment by Exagone313** [comment]
I think this issue should be renamed (Replace rather than Remove).

For GBA, there is an issue for replacing GBARunner2 by open_agb_firm: it currently uses a separate saves directory in the homebrew directory. I created this PR to be able to use saves in the same directory as the GBA rom: https://github.com/profi200/open_agb_firm/pull/209

If possible, I'd like to work on a solution to load the new open_agb_firm version through TWiLight, but don't know how it can work. Ideally this can be an option.

Important note: As the new save directory option I'm adding in open_agb_firm is not the default option, TWiLight would have to work around that as well.

**Comment by RocketRobz** [comment]
The emulators have been moved to the Virtual Console add-on some versions ago, so the emulators are now optional, and the guide already recommends to use the 3DS-mode emulators.

**Comment by Exagone313** [comment]
@RocketRobz Do you think it would be acceptable to contribute to TWiLight Menu++ to use [`open_agb_firm`](https://github.com/profi200/open_agb_firm) instead of GBARunner2 on 3DS? The [v27.15.0 release](https://github.com/DS-Homebrew/TWiLightMenu/releases/tag/v27.15.0) mentions the future [GBARunner3](https://github.com/Gericom/GBARunner3) but I don't think this project will use 3DS's hardware GBA support.

It would probably need a little fork of `open_agb_firm` to be able to run it seamlessly (to do not conflict with existing configuration). Both projects are GPLv3-licensed so it shouldn't be a problem shipping it as part of TWiLight Menu++.

PS: I have a lot of other projects so I will probably not work on it in the next months. I just want to know your stance on this topic since I had commented here earlier and you just closed this issue. I would be ok for contributing on this when I get the time.

EDIT: Apparently it's not possible to implement.

**Comment by RocketRobz** [comment]
> [@RocketRobz](https://github.com/RocketRobz) Do you think it would be acceptable to contribute to TWiLight Menu++ to use [`open_agb_firm`](https://github.com/profi200/open_agb_firm) instead of GBARunner2 on 3DS? The [v27.15.0 release](https://github.com/DS-Homebrew/TWiLightMenu/releases/tag/v27.15.0) mentions the future [GBARunner3](https://github.com/Gericom/GBARunner3) but I don't think this project will use 3DS's hardware GBA support.
> 
> It would probably need a little fork of `open_agb_firm` to be able to run it seamlessly (to do not conflict with existing configuration). Both projects are GPLv3-licensed so it shouldn't be a problem shipping it as part of TWiLight Menu++.
> 
> PS: I have a lot of other projects so I will probably not work on it in the next months. I just want to know your stance on this topic since I had commented here earlier and you just closed this issue. I would be ok for contributing on this when I get the time.
> 
> EDIT: Apparently it's not possible to implement.

Sorry, I missed your post.

For TWLMenu++ to use open_agb_firm to run GBA ROMs, there'd need to be a way for open_agb_firm to autoboot a GBA ROM file, like reading a ROM path from a `.ini` file like how flashcard kernels do it for DS ROMs.

**Comment by Exagone313** [comment]
> For TWLMenu++ to use open_agb_firm to run GBA ROMs, there'd need to be a way for open_agb_firm to autoboot a GBA ROM file, like reading a ROM path from a `.ini` file like how flashcard kernels do it for DS ROMs.

That part would be easy to implement. But is it possible for TWLMenu++ to trigger a reboot to a custom firmware in GBA mode?

**Comment by RocketRobz** [comment]
> That part would be easy to implement. But is it possible for TWLMenu++ to trigger a reboot to a custom firmware in GBA mode?

Yes. boot9strap has a feature where after a console reboot, it automatically starts `bootonce.firm` from the SD card root when detected.
TWLMenu++ could copy open_agb_firm to the SD root and rename it to `bootonce.firm` and then trigger a console reboot in order to boot open_agb_firm that way.

**Comment by Exagone313** [comment]
I think this has been added then reverted: https://github.com/SciresM/boot9strap/commit/ff16d59ff8fba431f5c5c934eea7db4d122eef1c

Luma3DS also rejected this feature.

I think I'm giving up entirely, it doesn't seem to be worth it to dig into this.
